kong/llm/drivers/shared.lua

```diff
@@ -301,77 +276,62 @@ function _M.post_request(conf, response_object)
   -- analytics and logging
   if conf.logging and conf.logging.log_statistics then
     local provider_name = conf.model.provider
+    local plugin_name = conf.__key__:match('plugins:(.-):')
```

Comment:
what if there are no matches? can you add some error handling here? some tests too?